mycpp/gc_str.cc

```diff
@@ -442,7 +442,7 @@ List<Str*>* Str::split(Str* sep) {
   int pos = 0;
   int end = n;
 
-  while (true) {
+  while (len(result) < max_split) {
```

Comment:
Hm one thing that is a bit suspicious is that there can be 2 append() to the list in a single iteration in some cases, i.e. with kEmptyString

Not sure if that actually causes a bug or not

I think this function was structured somewhat oddly, so I might poke at it a little more